test/apps/iodemo/io_demo.cc

```diff
@@ -1694,11 +1741,14 @@ class DemoClient : public P2pDemoCommon {
         for (size_t server_index = 0; server_index < _server_info.size();
              ++server_index) {
             LOG << "Disconnecting from " << server_name(server_index);
-            delete _server_info[server_index].conn;
-            _server_info[server_index].conn = NULL;
+            new Disconnector(*this, *_server_info[server_index].conn);
+        }
+
+        while (!_active_servers.empty()) {
```

Comment:
add log message before: if list not empty smth like this
`UCX_LOG << "Waiting for " << num << " disconnects to complete"`